pyterrier/batchretrieve.py

```diff
@@ -570,6 +620,14 @@ def __setstate__(self, d):
         self.properties.update(d["properties"])
         for key,value in d["properties"].items():
             self.appSetup.setProperty(key, str(value))
+        #TODO consider the context state?
+
+    @staticmethod 
+    def from_dataset(dataset : Union[str,Dataset], 
```

Comment:
looks like this belongs in a different PR?

Reply:
I started from a from_dataset branch, but I think this bit is already merged.